### PR TITLE
Add empty line utils

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #3525 

just has the utils

### CODE REVIEW

File: src/utils/parser/getEmptyLines.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  // The following sequence stores lines which have executable code
+  // (contents other than comments or EOF, regardless of line position)
+  const executableLines = ast.tokens
+    .filter(token => {
+      return (
+        !commentTokens.includes(token.type) &&
+        (!token.type || (token.type.label && token.type.label != "eof"))
+      );
+    })
```
Comment (jasonLaster) [inline src/utils/parser/getEmptyLines.js]: small nit, but we can remove the return here

```js
.filter(token => (
   !commentTokens.includes(token.type) &&
   (!token.type || (token.type.label && token.type.label != "eof"))
))
```

File: src/utils/parser/getEmptyLines.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    .map(token => token.loc.start.line - 1);
+  executableLines = uniq(executableLines);
+
+  // Populates a pre-filled array of every line number,
+  // then removes lines which were found to be executable
+  const fillRange = (start, end) => {
+    return Array(end - start + 1).fill().map((item, index) => start + index);
+  };
```
Comment (jasonLaster) [inline src/utils/parser/getEmptyLines.js]:
```js
const fillRange = (start, end) => (
 Array(end - start + 1).fill().map((item, index) => start + index);
))
```

Comment (jasonLaster) [reply src/utils/parser/getEmptyLines.js]: would `fillRange` work here?

**Comment by darkwing** [comment]
Wow, this looks excellent!

**Pull request merged**